### PR TITLE
feat(Connector): [Stripe] Implement Cashapp Wallet 

### PR DESCRIPTION
**Pull request by AkshayaFoiger**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Implemented Cashapp wallet, payment method for Stripe.

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
In Stripe connector Cashapp was not implemented, but it is important payment method which is needed in market so I implemented it.

## How did you test it?
Screen Shot 2023-08-01 at 5 00 44 PM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/api_models/src/payments.rs [hunk 1]
```diff
@@ -711,6 +711,8 @@ pub enum WalletData {
     PaypalSdk(PayPalWalletData),
     /// The wallet data for WeChat Pay Redirection
     WeChatPayRedirect(Box<WeChatPayRedirection>),
+    //The wallet data for Cashapp
+    Cashapp(Box<CashAppRedirection>),
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:715]: Nit: Please follow consistent naming, one of `Cashapp` or `CashApp` throughout the codebase.

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

**Comment by ArjunKarthik** [comment]
@AkshayaFoiger As discussed right now we don't support mobile app redirection flow or QR code flow. So removing this PR from waiting for review

**Comment by ArjunKarthik** [comment]
@AkshayaFoiger Since we have solution for handling QR redirects and QR mobile redirects. Please implement the QR code handling for this payment method

File: crates/router/build.rs [hunk 2]
```diff
@@ -2,7 +2,7 @@ fn main() {
     // Set thread stack size to 4 MiB for debug builds
     // Reference: https://doc.rust-lang.org/std/thread/#stack-size
     #[cfg(debug_assertions)]
-    println!("cargo:rustc-env=RUST_MIN_STACK=4194304"); // 4 * 1024 * 1024 = 4 MiB
+    println!("cargo:rustc-env=RUST_MIN_STACK=6194304"); // 4 * 1024 * 1024 = 4 MiB
```
Comment (ArjunKarthik) [inline crates/router/build.rs]: Remove this

**Review by ArjunKarthik — DISMISSED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by jagan-jaya — APPROVED** [review]

**Pull request merged**